### PR TITLE
[TASK] Add documentation generation info for MacOS

### PR DESCRIPTION
**Pull request by afoeder**
Besides, this adds some formatting and improves sectioning.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+
+### Using Ubuntu 14.04 LTS:
 
 1. Run ./bin/install-dependencies.sh
 2. Run ./bin/generate-docs.sh
 
-It will generate the documentation into the build directory of the checkout.
+### Using macOS:
+
+Sphinx Installation instructions can be found at the
```
Comment (Ocramius) [inline docs/README.md]: This is no longer needed with the new website, tbh /cc @jwage

Comment (afoeder) [reply docs/README.md]: ah nice even better; just stumbled upon this when I wanted to verify #7185 and wanted to share my insights :) I realized that it gets rendered in the old-school style, but for veryfing the linking it was fair enough.

Comment (jwage) [reply docs/README.md]: This is updated here https://github.com/doctrine/doctrine2/pull/7177

Comment (afoeder) [reply docs/README.md]: ok, I guess mine can be closes then. Thanks!

**Pull request closed without merging**